eth-types/src/evm_types/stack.rs

```diff
@@ -128,8 +128,11 @@ impl Stack {
         self.0.last().cloned().ok_or(Error::InvalidStackPointer)
     }
 
-    /// Returns the second last [`Word`] allocated in the `Stack`.
+    /// Returns the n-th last [`Word`] allocated in the `Stack`.
     pub fn nth_last(&self, nth: usize) -> Result<Word, Error> {
+        if self.0.len() < (nth + 1) {
+            return Err(Error::InvalidStackPointer);
+        }
```

Comment:
Was wondering why we need to check the index again if we already using `get(...).or_ok(...)`, then realized it will cause integer underflow in line 137.